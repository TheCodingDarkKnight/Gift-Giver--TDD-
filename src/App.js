import React, { Component } from "react";
import { Button } from "react-bootstrap";

import Gift from "./Gift";

class App extends Component {
    state = {
        gifts: []
    }

    addGift = (event) => {
        const { gifts } = this.state;
        const max_id = gifts[gifts.length - 1] ? gifts[gifts.length - 1].id : 0;

        this.setState({ gifts: [...gifts, { id: max_id + 1 }] })
    }

    render() {
        const { gifts } = this.state;
        return (
            <div>
                <h2>Gift Giver</h2>
                <ul className="gift-list">
                    {
                        gifts.map(({ id }) => <Gift key={id} />)
                    }
                </ul>
                <Button className="btn-add" onClick={this.addGift}>Add Gift</Button>
            </div>
        )
    }
}

export default App;
