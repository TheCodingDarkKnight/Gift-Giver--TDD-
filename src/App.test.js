import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import App from "./App";

configure({ adapter: new Adapter() });

const app = shallow(<App />);

it("Render App correctly", () => {
    expect(app).toMatchSnapshot();
})