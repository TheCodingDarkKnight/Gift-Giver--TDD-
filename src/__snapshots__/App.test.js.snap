// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App correctly 1`] = `ShallowWrapper {}`;
